### ISSUE TITLE
Missing `depends_on` in NeXus objects

### ISSUE THREAD

**Issue by g5t** [mducle/eniius#11]
## Description
When converting a `McStasScript` object to `NeXus`, the positions and orientations of instrument components are determined as one or more linked affine transformations relative to the sample position.
The transformation(s) are then added to `NX*` base class as a `NXtransformations` group called `'transforms'`.

Compared to the description of `depends_on` in the [NeXus Design documentation](https://manual.nexusformat.org/design.html#coordinate-transformation-field-and-attributes), the `eniius` implementation is missing a `depends_on` value in the base class, which in this case should be `'transforms/{outermost-transformation-name}'`. 

Quoting from the linked source:
> For example, consider an eulerian cradle as used on a four-circle diffractometer. Such a cradle has a dependency chain of `phi:chi:rotation_angle`. Then the `depends_on` field in [NXsample](https://manual.nexusformat.org/classes/base_classes/NXsample.html#nxsample) would have the value `phi`.

With a useful diagram: 
```  sample:NXsample
    transforms:NXtransformations
      rotation_angle
        @transformation_type=rotation
        @vector=0,1,0
        @offset=0,0,0
        @depends_on=.
      chi
        @transformation_type=rotation
        @vector=0,0,1
        @offset=0,0,0
        @depends_on=rotation_angle
      phi
        @transformation_type=rotation
        @vector=0,1,0
        @offset=0,0,0
        @depends_on=chi
    depends_on 
      transforms/phi
```

## Why this is a problem
Without the `depends_on` field, components represented as `NX*` base class objects can not be positioned correctly in the instrument _without extrernal knowledge_. Therefore, external libraries like [nexus-constructor](https://github.com/ess-dmsc/nexus-constructor/) can not be used with `eniius` output.

**Comment by g5t** [comment]
Commit https://github.com/g5t/eniius/commit/9e526a165bc3b2ded5f4d4eb69af5572814bb2bc, part of #10, adds relative `depends_on` children to `NX*` objects during translation from `McStas` component-form. This (seems to) conform to the NeXus standard, but is not understood by, e.g., `nexus-constructor`.

**Comment by g5t** [comment]
The merged #10 includes optional absolute `depends_on` resolution; so this issue has been resolved.
